src/relay/pass/pattern_util.h

```diff
@@ -100,11 +101,31 @@ inline Expr ExpandBiasToMatchAxis(Expr bias,
   return bias;
 }
 
+/*!
+ * \brief Check if the call is depthwise conv2d.
+ *
+ * \param call The conv2d call.
+ * \param param The conv2d attributes.
+ * \return Whether it is depthwise_conv2d.
+ */
+inline bool IsDepthwiseConv2D(const Call& call,
+                              const Conv2DAttrs* param,
+                              const Layout& weight_layout) {
+  static const Layout kOIHW("OIHW");
+  auto wshape = ConvertLayout(
+      call->args[1]->type_as<TensorTypeNode>()->shape,
+      weight_layout, kOIHW);
+  return is_const_int(wshape[0], param->groups) &&
+      is_const_int(wshape[1], 1);
```

Comment:
Could we use ```is_const_int(wshape[0], param->groups)``` to check depthwise convolution? In theory, wshape[1] could be not equal to 1, although TF / MXNet / Caffe and so on frontends will make it be 1 even though the depth channel multiplier not  be 1 (such as 0.25)

Reply:
The current code makes the assumption of input and output groups being even, and the special logic relies on this

 A better approach would enhance the Conv2D code itself to handle general group convolution, which should not be too hard as a followup contribution from anyone in the community:)